DataMgr/ForeignStorage/ArrowForeignStorage.cpp

```diff
@@ -105,138 +100,160 @@ class ArrowForeignStorageBase : public PersistentForeignStorageInterface {
+
+  auto resultBuf =
+      arrow::AllocateBuffer(sizeof(T) * arr_col_chunked_array->length()).ValueOrDie();
+  auto resultData = reinterpret_cast<T*>(resultBuf->mutable_data());
 
   tbb::parallel_for(
-      tbb::blocked_range<size_t>(0, fragments.size()),
-      [&](const tbb::blocked_range<size_t>& r0) {
-        for (size_t f = r0.begin(); f != r0.end(); ++f) {
-          tbb::parallel_for(
-              tbb::blocked_range<size_t>(fragments[f].first_chunk,
-                                         fragments[f].last_chunk + 1),
-              [&](const tbb::blocked_range<size_t>& r1) {
-                for (auto chunk_index = r1.begin(); chunk_index != r1.end();
-                     ++chunk_index) {
-                  auto chunk = arr_col_chunked_array->chunk(chunk_index).get();
-                  if (chunk->data()->null_count == chunk->data()->length) {
-                    // it means we will insert sentinel values in read function
-                    continue;
-                  }
-                  // We can not use mutable_data in case of shared access
-                  // This is not realy safe, but it is the only way to do this without
-                  // copiing
-                  // TODO: add support for sentinel values to read_csv
-                  auto data = const_cast<uint8_t*>(chunk->data()->buffers[1]->data());
-                  if (data && chunk->null_bitmap()) {  // TODO: to be checked and possibly
-                                                       // reimplemented
-                    // CHECK(data) << " is null";
-                    T* dataT = reinterpret_cast<T*>(data);
-                    const uint8_t* bitmap_data = chunk->null_bitmap_data();
-                    const int64_t length = chunk->length();
-                    const int64_t bitmap_length = chunk->null_bitmap()->size() - 1;
-
-                    for (int64_t bitmap_idx = 0; bitmap_idx < bitmap_length;
-                         ++bitmap_idx) {
-                      T* res = dataT + bitmap_idx * 8;
-                      for (int8_t bitmap_offset = 0; bitmap_offset < 8; ++bitmap_offset) {
-                        auto is_null = (~bitmap_data[bitmap_idx] >> bitmap_offset) & 1;
-                        auto val = is_null ? null_value : res[bitmap_offset];
-                        res[bitmap_offset] = val;
-                      }
-                    }
-
-                    for (int64_t j = bitmap_length * 8; j < length; ++j) {
-                      auto is_null = (~bitmap_data[bitmap_length] >> (j % 8)) & 1;
-                      auto val = is_null ? null_value : dataT[j];
-                      dataT[j] = val;
-                    }
-                  }
-                }
-              });
+      tbb::blocked_range<size_t>(0, arr_col_chunked_array->num_chunks()),
+      [&](const tbb::blocked_range<size_t>& r) {
+        for (size_t c = r.begin(); c != r.end(); ++c) {
+          size_t offset = 0;
+          for (int i = 0; i < c; i++) {
```

Comment:
Update `i` to `size_t` in order to avoid compiler warning.

Reply:
fixed